### ISSUE TITLE
All my actors became detached

### ISSUE THREAD

**Issue by gcolgate** [ggagnon76/token-independence#28]
1. Have a scene with 155K worth of actors.
2. Use Token Independence to embed all the actors in the scene. Note that the actors are in an exported version of the scene.
3. Delete all the actors from the world
4. Everything still works great
5. Shut down foundry
6. Restart foundry on a new day. All the actors are detached and are no longer in the scene embedded, since they are deleted they no longer exist

Perhaps I misunderstood the purpose of Token Independence. If you have any time you can email me at gcolgate@gmail.com

**Comment by ggagnon76** [comment]
Hello.
When I first submitted this module, it was rejected because modules that try to store too much in flags can cause database corruption.  To be able to release my module, I had to add a check that limited how much data is allowed to be stored.  

With 155k actors, I'm going to go out on a limb and say you exceeded the 1mb limit.  You should have gotten a notification that the embed operation failed, ie: "The selected actors exceeded the 1mb limit!"

The intent of my module is to do what you think it is supposed to do.  But 155k is... well it blows my mind that you had that many actors.  I mean, if it takes 10 seconds to create an actor, it would take 400+ hours to create that many...

I will absolutely fix any bugs that are found.  However, in this situation, I believe you are trying to use Foundry in a way no one could have fathomed.

**Comment by ggagnon76** [comment]
Latest version implements a warning dialog to clearly inform users when actors have NOT been embedded because the flag allotment has been exceeded.

**Comment by gcolgate** [comment]
Not 155k actors, the size of the actors were 155k BYTES total. For some reason they can be quite large, the scene had about 200 actors in it.

**Comment by ggagnon76** [comment]
AHH.  Well that makes much more sense.
I did find a 'bug' while working on this version.  If the user changed the name of the token in the configuration (not the character sheet), then it would break the token.  (Users may do this to display the correct name in the combat tracker)

I fixed the above issue by placing the name of the actor as a flag on the token on token creation, so I don't have to rely on that user editable field to identify what tokens go with what actors.

If you can confirm that you did edit that token config name on those 200 actors (That's still a lot!  Did you mean 200 tokens?), then I am confident that version 0.2.0 will have resolved your issue.

I would appreciate it if you could test?

**Comment by gcolgate** [comment]
Token Mold had renamed all the characters. So that could be the bug. For example, a particular token,  Insistent Gauntling, the actor is Gauntling. All the tokens had new names.

**Comment by ggagnon76** [comment]
That is possible.  If you created the tokens with Token Mold before enabling Token Independence, then everything would have been broken.

I realized this was a possibility when I refactored for V9 and fixed that by relying on a flag instead.  Token Mold can now change both token.name and token.actorData.name and it won't affect Token Independence any longer.  However, this does mean that any tokens created when Token Independence is not enabled won't work with Token Independence.  Unfortunate, but I saw no other way of preventing users from breaking the 'link' between token and embedded actor.

I'm sorry you ran into those issues.  If you continue to use this module, and upgrade to 0.2.1, Token Independence will now work properly with Token Mold.
